src/common.h

```diff
@@ -93,6 +93,13 @@ void *strophe_realloc(const xmpp_ctx_t *ctx, void *p, size_t size);
 char *strophe_strdup(const xmpp_ctx_t *ctx, const char *s);
 char *strophe_strndup(const xmpp_ctx_t *ctx, const char *s, size_t len);
 void strophe_free(const xmpp_ctx_t *ctx, void *p);
+#define strophe_free_and_null(ctx, p) \
```

Comment:
I thought a macro would do the job.

Giving back the question: Why a function and not just a macro?

Reply:
> I thought a macro would do the job.
> 
> Giving back the question: Why a function and not just a macro?

You can't use macro as a pointer and in future if you want to create some dynamic object with dynamic free'ing, it's better to have this as a function instead of macro.

Reply:
OK. Then we'll change it to a macro in case we ever need it. It's only internal API so nobody will notice.